src/ec/packages/ec-component-radio/radio-group.html.twig

```diff
@@ -59,44 +56,48 @@
 
 {# Print the result #}
 
-<div
+<fieldset
   class="{{ _css_class }}"
-  role="radiogroup"
-  aria-labelledby="{{ label_id }}"
-  {{ _helper_id is not empty ? "aria-describedby=\"" ~ _helper_id ~ "\""}}
+  {{ _helper_id is not empty ? 'aria-describedby="' ~ _helper_id ~ '"'}}
   {{ _extra_attributes|raw }}
 >
 
   {%- if _label is not empty %}
-    <div
-      class="ecl-form-label{{ hide_label ? ' ecl-form-label--hidden'}}"
-      id="{{ _label_id }}"
+    <legend
+      class="ecl-form-label
+      {{- _invalid ? ' ecl-form-label--invalid' }}"
     >
       {%- block label _label -%}
-    </div>
+      {%- if _required and _required_text is not empty %}
+        <span class="ecl-form-label__required">{{ _required_text }}</span>
+      {%- elseif _optional_text is not empty %}
```

Comment:
But it's not even like this, @papegaill :) 
Now we are potentially saying that if it's required but it does not have any text associated with that, we output anyway the optional span if the optional text is set...
So, to recap:
- we print the span for the mandatory group when the group is required and the text is set
- we print the span for the optional when the group is not required and the optional text is not empty
ca va..? :)

Reply:
ca va bien :)